pylint/checkers/classes.py

```diff
@@ -1000,11 +1000,15 @@ def _check_unused_private_attributes(self, node: nodes.ClassDef) -> None:
                 if attribute.attrname != assign_attr.attrname:
                     continue
 
-                if assign_attr.expr.name == "cls" and attribute.expr.name in [
-                    "cls",
-                    "self",
-                ]:
-                    # If assigned to cls.attrib, can be accessed by cls/self
+                if (
+                    assign_attr.expr.name
+                    in [
+                        "cls",
+                        node.name,
+                    ]
```

Comment:
Why do we not have "self" in this list, and shouldn't we create a variable ``this_class_names =  ["cls", "self", node.name]``  ? Also, is this compatible with https://github.com/PyCQA/pylint/pull/4949#pullrequestreview-746579519 ?

Reply:
if assignment is done to attributes of `cls` or the class name itself, we can access it via `cls`, `self` and the class name.

however, if it's assigned to `self`, then it is only accessible by `self`